src/tools/wasm-reduce.cpp

```diff
@@ -854,18 +854,14 @@ struct Reducer
       reduceByZeroing(
         segment.get(),
         first,
-        [&](Expression* entry) {
-          if (entry->is<RefNull>()) {
-            // we don't need to replace a ref.null
+        [&](Expression* elem) {
+          if (elem->is<RefNull>()) {
+            // We don't need to replace a ref.null.
             return true;
-          } else if (first->is<RefNull>()) {
-            return false;
-          } else {
-            // Both are ref.func
-            auto* f = first->cast<RefFunc>();
-            auto* e = entry->cast<RefFunc>();
-            return f->func == e->func;
           }
+          // Replace the element if it is different from our first "zero"
+          // element.
+          return !ExpressionAnalyzer::equal(first, elem);
```

Comment:
This looks flipped? Before we returned `f->func == e->func` which meant we return true when they were equal.

Reply:
Fixed!